internal/sdkprovider/service/organization/organization_project_test.go

```diff
@@ -264,66 +240,326 @@ func TestAccAivenOrganizationProjectUpdateStages(t *testing.T) {
+  tag {
+    key   = "key1"
+    value = "value1"
+  }
+  tag {
+    key   = "key2"
+    value = "value2"
+  }
+}
+`, aivenOrganizationProjectResource, projectID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
+					resource.TestCheckResourceAttrPair(resourceName, "organization_id", "aiven_organization.foo", "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "billing_group_id", "aiven_billing_group.foo", "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "parent_id", "aiven_organizational_unit.foo", "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "ca_cert"),
+					resource.TestCheckResourceAttr(resourceName, "tag.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "tag.*", map[string]string{
+						"key":   "key1",
+						"value": "value1",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "tag.*", map[string]string{
+						"key":   "key2",
+						"value": "value2",
+					}),
+				),
+			},
+			{
+				// test the data source
+				Config: baseConfig + fmt.Sprintf(`
+resource "%s" "foo" {
+  project_id       = "%s"
+  organization_id  = aiven_organization.foo.id
+  billing_group_id = aiven_billing_group.foo.id
+  parent_id        = aiven_organizational_unit.foo.id
+  tag {
+    key   = "key1"
+    value = "value1"
+  }
+  tag {
+    key   = "key2"
+    value = "value2"
+  }
+}
+
+data "%s" "foo" {
+  project_id      = aiven_organization_project.foo.project_id
+  organization_id = aiven_organization_project.foo.organization_id
+}
+`, aivenOrganizationProjectResource, projectID, aivenOrganizationProjectResource),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "project_id", projectID),
+					resource.TestCheckResourceAttrPair(dataSourceName, "organization_id", "aiven_organization.foo", "id"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "billing_group_id", "aiven_billing_group.foo", "id"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "parent_id", "aiven_organizational_unit.foo", "id"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "ca_cert"),
+					resource.TestCheckResourceAttr(dataSourceName, "tag.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(dataSourceName, "tag.*", map[string]string{
+						"key":   "key1",
+						"value": "value1",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(dataSourceName, "tag.*", map[string]string{
+						"key":   "key2",
+						"value": "value2",
+					}),
 				),
 			},
 		},
 	})
 }
 
+// generateBaseConfig creates the base configuration for organizations, billing groups and organizational units
+func generateBaseConfig(rName string) string {
+	return fmt.Sprintf(`
+# foo organization
+resource "aiven_organization" "foo" {
+  name = "test-acc-org-%[1]s"
+}
+
+# bar organization
+resource "aiven_organization" "bar" {
+  name = "test-acc-org-%[1]s-bar"
+}
```

Comment:
The comment says bar organization is organization bar. Probably it was generated, but it doesn't make sense to have it. Same for the rest: I read them, go thoroughly through, yet they say things I can tell from this little peace of code.